<?xml version="1.0"?>
<launch>
	
	<!--<node pkg="omni3ros_pkg" type="teleop_twist_keyboard.py" name="teleop_twist_keyboard" output="screen">
	</node>-->
	
	  <!-- convert provide equal cmd_vel to both motors-->
  <node pkg="omni3ros_pkg" type="twist_motor_omni.py" name="twist_motor" output="screen"/>
  <!-- Publish Odom and Broadcast TF-->
  <!--<node pkg="model_control" type="diff_tf.py" name="diff_tf" output="screen"/>--> 
  <!-- PID control and generate PWM for motors-->
	
	<node name="lpid_velocity" pkg="omni3ros_pkg" type="pid_velocity_omni.py" output="screen">
      <remap from="wheel" to="lwheel_encoder"/>
      <remap from="motor_cmd" to="lmotor_cmd"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="lwheel_vel"/>
      <rosparam param="Kp">200</rosparam>
      <rosparam param="Ki">15</rosparam>
      <rosparam param="Kd">0.001</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">100</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">100</rosparam>
      <rosparam param="ticks_meter">31828</rosparam>
  </node>

  <node name="rpid_velocity" pkg="omni3ros_pkg" type="pid_velocity_omni.py" output="screen">
      <remap from="wheel" to="rwheel_encoder"/>
      <remap from="motor_cmd" to="rmotor_cmd"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="rwheel_vel"/>
      <rosparam param="Kp">200</rosparam>
      <rosparam param="Ki">15</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">100</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">100</rosparam>
      <rosparam param="ticks_meter">31828</rosparam>
  </node>
  
    <node name="bpid_velocity" pkg="omni3ros_pkg" type="pid_velocity_omni.py" output="screen">
      <remap from="wheel" to="bwheel_encoder"/>
      <remap from="motor_cmd" to="bmotor_cmd"/>
      <remap from="wheel_vtarget" to="bwheel_vtarget"/>
      <remap from="wheel_vel" to="bwheel_vel"/>
      <rosparam param="Kp">200</rosparam>
      <rosparam param="Ki">15</rosparam>
      <rosparam param="Kd">0.04</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">100</rosparam>
      <rosparam param="timeout_ticks">10</rosparam>
      <rosparam param="rolling_pts">100</rosparam>
      <rosparam param="ticks_meter">31828</rosparam>
  </node>

  <node pkg="robotino_ros" type="command.py" name="command" output="screen"/>

	
</launch>
